### PR TITLE
[webnn] Add tests for WebNN API cast op

### PR DESCRIPTION
**Pull request by BruceDai**
Current this pr doesn't include tests of uint64 type which will be added later.
@fdwr @Honry PTAL, thanks!

### CODE REVIEW

File: webnn/resources/test_data/cast.json [hunk 1]
```diff
@@ -0,0 +1,3040 @@
...
+        "type": "int32"
+      }
+    },
+    {
+      "name": "cast float32 1D tensor to int32",
+      "inputs": {
+        "input": {
+          "shape": [24],
+          "data": [
+            102.1578369140625,
```
Comment (fdwr) [inline webnn/resources/test_data/cast.json:30]: We should include some negative values too, one to test truncating down toward negative infinity, one toward zero, and half decimal. e.g. [-4.9, -5.2, 6.5]

File: webnn/resources/test_data/cast.json [hunk 2]
```diff
@@ -0,0 +1,3040 @@
...
+        "type": "int64"
+      }
+    },
+    {
+      "name": "cast uint32 4D tensor to int8",
+      "inputs": {
+        "input": {
+          "shape": [2, 2, 2, 3],
+          "data": [
+            34,
```
Comment (fdwr) [inline webnn/resources/test_data/cast.json:1705]: At least one negative value would be prudent.

File: webnn/resources/utils.js [hunk 3]
```diff
@@ -289,6 +289,7 @@ const PrecisionMetrics = {
   argMax: {ULP: {int64: 0}},
   argMin: {ULP: {int64: 0}},
   batchNormalization: {ULP: {float32: 6, float16: 6}},
+  cast: {ULP: {float32: 1, float16: 1, int32: 0, uint32: 0, int64: 0, int8: 0, uint8: 0}},
```
Comment (fdwr) [inline webnn/resources/utils.js:292]: These are good. Cast is interesting because the ULP *technically* depends on combination of input and the output data types, but I don't know that you can easily represent that via this lookup table (which is why I put 1 ULP in my table).

- int8 -> int32 : 0 ULP
- int32 -> int8 : 0 ULP (assuming value is between -128 and 127)
- int32 -> float32 : 0 ULP (assuming abs(x) <= 16777216)
- float32 -> int32 : 0 ULP
- float16 -> float32 : 0 ULP
- float32 -> float16 : 1 ULP
- float64 -> float32 : 1 ULP (not that WebNN supports float64 anyway)

Ideally we could just say 0 ULP for all cases, but some GPU's round toward zero, while others round toward nearest even. Round to zero is the [D3D specified mode](https://microsoft.github.io/DirectX-Specs/d3d/archive/D3D11_3_FunctionalSpec.htm#3.2.2%20Floating%20Point%20Conversion), but not all GPU's do so. (we hit this issue for float64 to float32 casting).

**Review by fdwr — APPROVED** [review]
Some negative values would be good. Otherwise LGTM. Thanks Bruce.

File: webnn/resources/test_data/cast.json (hunk 2 above)
Comment (BruceDai) [reply webnn/resources/test_data/cast.json:1705]: The data of **uint32** type couldn't be a negative number. Do you suggest test negative data for other casting? Thanks.

**Review by fdwr — APPROVED** [review]
👍

**Review by Honry — APPROVED** [review]
LGTM, thanks!
